pkg/arvo/sys/hoon.hoon

```diff
@@ -11534,114 +11557,175 @@
+        ::of a prefix comment instead of postfix
+        ::;~(less (exit ;~(plug (plus en-link) col ace)) (exit line))
         (exit line)
       ::
-        (easy *whit)
+        (easy *help)
       ==
     ::
-    ::
-    ++  beer
-      |=  $:  lab=(unit term)
-              boy=(unit (pair cord (list sect)))
-              def=(list (pair (pair term cord) (list sect)))
+    ++  leap                                            ::  whitespace less docs
+      %+  cold  ~
+      %-  plus
+      ;~  plug
+        (star ace)
+        ;~  pose
+          ;~(plug (just '\0a') (star ace))
+          ;~  plug  col  col
+            ;~(less ;~(pose larg smol) ;~(plug (star prn) (just '\0a')))
           ==
-      ^-  whit
-      =;  def  [lab boy (malt def) ~]
-      (turn def |=([[a=term b=cord] c=(list sect)] [a [b c]]))
-    ::
-    ::
-    ++  body
-      ;~  pose
-        ;~  plug                                        :: can duplicate ::
-          (into ;~(pfix (punt ;~(plug null col gar step)) line))
-          (easy ~)
         ==
+      ==
+    ::
+    ::  +smol: 2 aces then summary, 4 aces then paragraphs.
+    ++  smol
+      ;~  pfix
+        step
         ;~  plug
-          (into ;~(pfix step line))
-          (rant text)
+          ;~  plug
+            (plus en-link)
+            ;~(sfix ;~(pfix col ace (cook crip (star prn))) (just '\0a'))
+          ==
+          ;~(pfix null (rant ;~(pfix step step text)))
         ==
       ==
     ::
-    ++  text  (pick line code)                          ::  text line
-    ++  line  ;~(less ace (cook crip (star prn)))       ::  prose line
-    ++  code  ;~(pfix step step (cook crip (star prn))) ::  code line
-    ++  noel  ;~(plug (punt ;~(pfix step hax)) null)    ::  header padding
-    ++  head  ;~(pfix hax step cen sym)                 ::  header line
-    ++  null  (cold ~ (star ace))                       ::  blank line
-    ++  fine                                            ::  definition line
-      ;~  (glue ;~(plug col ace))
-        sym
-        (cook crip (star prn))
+    ::  +larg: 4 aces then summary, 2 aces then paragraphs.
+    ++  larg
+      ;~  pfix
+        step  step
+        ;~  plug
+          ;~  sfix
+            ;~  plug
+              ;~  pose
+                ;~(sfix (plus en-link) col ace)
+                ;~(less ace (easy *cuff))
+              ==
+              (cook crip (star prn))
+            ==
+            (just '\0a')
+          ==
+          ;~(pfix null (rant ;~(pfix step teyt)))
+        ==
       ==
     ::
+    ++  myth
+      |*  bod=rule
+      (cook |=([[a=cuff b=cord] c=(list sect)] [a b c]) bod)
     ::
-    ::  step: indent
-    ::  into: :: and indent to end of line, consuming following space.
-    ::  indo: :: to end of line, consuming following space.
-    ::  exit: :: to end of line, not consuming following space.
+    ++  rant
+      |*  sec=rule
+      %-  star
+      ;~  pfix
+        (into null)
+        (plus (into sec))
+      ==
```

Comment:
Likewise, this and its two callsites feel like a somewhat poor generalization. Can you describe what `+rant` does in a short sentence? If not, may be more clear to just duplicate its logic instead.

What particularly jumps out to me is how it gets called with parsers that start by eating a `step` (or two), even though `into` eats as many `ace`s as it can at the end. ...And `into` is only used here afaict.

Reply:
Yeah there's probably some better factoring to do here. I'll give it another look.

Reply:
I refactored this a little bit, but didn't remove `+rant`. Even if its only used twice, its creating a `(list sect)` which is one of the central structures in doccords, so I think it deserves its own arm for readability.

https://github.com/urbit/urbit/pull/5873/commits/ca3140a8b3997dd08976f8c09bfd1bdeebd4b14e